dts/bindings/sensor/ti,tmp116.yaml

```diff
@@ -8,3 +8,20 @@ compatible: "ti,tmp116"
 bus: tmp116
 
 include: [sensor-device.yaml, i2c-device.yaml]
+
+properties:
+  sample-frequency:
+    type: int
+    default: 1000000
+    description: |
+      Sample frequency in micro hertz.
+      Default is power-up configuration.
+    enum:
+      - 64000000
+      - 8000000
+      - 4000000
+      - 2000000
+      - 1000000
+      - 250000
+      - 125000
+      - 62500
```

Comment:
Consider implementing the ENUMs using macro-define dt-bindings so they're referenced both in DTS and C (e.g: used for ICM42688):
- DT Bindings header:
https://github.com/zephyrproject-rtos/zephyr/blob/18fb3f9a329ade44495983206a5263f7aa2a84a4/include/zephyr/dt-bindings/sensor/icm42688.h
- DTS usage:
https://github.com/zephyrproject-rtos/zephyr/blob/18fb3f9a329ade44495983206a5263f7aa2a84a4/boards/tdk/robokit1/robokit1-common.dtsi#L109
- Source code usage:
https://github.com/zephyrproject-rtos/zephyr/blob/18fb3f9a329ade44495983206a5263f7aa2a84a4/drivers/sensor/tdk/icm42688/icm42688.h#L99-L132

Reply:
Thanks for the suggestion @ubieda . I gave it a shot. Let me know what you think.